### PR TITLE
Add smartmotion and arbitrary feedforward to intake deploy and arm

### PR DESCRIPTION
**Pull request by Panda8188**
* Calculated offset of the intake and arm off of their horizontal
* Made placeholder variables of arbitrary intake and arm feed forward constants
* Made a PID slot variable
* Added arbitrary feed forward to intake and lift `setReference` functions

### CODE REVIEW

File: src/main/java/frc/robot/Calibrations.java [hunk 1]
```diff
@@ -82,4 +82,9 @@ public static final class SwerveModuleCalibrations {
 
   /** High profile constraints = pure P controller */
   public static final PIDController PATH_THETA_CONTROLLER = new PIDController(9.0, 0.0, 0.80);
+
+  public static final double ARBITRARY_ARM_FEED_FORWARD = PLACEHOLDER_DOUBLE;
+  public static final double ARBITRARY_INTAKE_FEED_FORWARD = PLACEHOLDER_DOUBLE;
```
Comment (balghane) [inline src/main/java/frc/robot/Calibrations.java]: Add units and comments

File: src/main/java/frc/robot/Calibrations.java [hunk 2]
```diff
@@ -82,4 +82,9 @@ public static final class SwerveModuleCalibrations {
...
+
+  public static final int SMART_MOTION_SLOT = 0;
```
Comment (balghane) [inline src/main/java/frc/robot/Calibrations.java]: I would define this in each subsystem separately

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 3]
```diff
@@ -63,7 +63,10 @@ public Intake() {
   /** Deploys the intake out */
   public void deploy() {
     deployMotorPID.setReference(
-        Calibrations.INTAKE_DEPLOYED_POSITION_DEGREES, CANSparkMax.ControlType.kPosition);
+        Calibrations.INTAKE_DEPLOYED_POSITION_DEGREES,
+        CANSparkMax.ControlType.kPosition,
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Intake.java]: This should be kSmartMotion or something like that

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 4]
```diff
@@ -63,7 +63,10 @@ public Intake() {
...
+        Calibrations.SMART_MOTION_SLOT,
+        Calibrations.ARBITRARY_INTAKE_FEED_FORWARD * getCosineIntakeAngle());
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Intake.java]: Use version of the function where we specify Volts as the feedforward units

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 5]
```diff
@@ -137,6 +142,11 @@ public boolean doneScoring() {
     return true; // Placeholder until logic is made
   }
 
+  /** Returns the cosine of the arm angle in degrees off of the horizontal. */
+  public double getCosineArmAngle() {
+    return Math.cos(armEncoder.getPosition() - 90);
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: This 90 and the 180 in the other subsystem are "magic numbers." It's not obvious why the number is what it is. Should be a named variable that makes it clear what it's doing

**Review by balghane — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 6]
```diff
@@ -63,7 +66,11 @@ public Intake() {
   /** Deploys the intake out */
   public void deploy() {
     deployMotorPID.setReference(
-        Calibrations.INTAKE_DEPLOYED_POSITION_DEGREES, CANSparkMax.ControlType.kPosition);
+        Calibrations.INTAKE_DEPLOYED_POSITION_DEGREES,
+        CANSparkMax.ControlType.kSmartMotion,
+        SMART_MOTION_SLOT,
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Intake.java:80]: Need to configure smartmotion stuff for intake and arm motors

**Review by tails618 — APPROVED** [review]

**Review by balghane — APPROVED** [review]

**Comment by tails618** [comment]
/forma
edit: oops

**Pull request merged**